tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/AbstractHelmChartIT.java

```diff
@@ -1,92 +1,178 @@
+    protected final @NotNull String operatorNamespace = getOperatorNamespaceName(getClass());
 
-    @SuppressWarnings("NotNullFieldNotInitialized")
-    protected @NotNull KubernetesClient client;
+    @BeforeAll
+    @Timeout(value = 5, unit = TimeUnit.MINUTES)
+    static void baseBeforeAll() {
+        Awaitility.setDefaultPollInterval(Duration.ofSeconds(3));
+        Awaitility.setDefaultTimeout(Duration.ofMinutes(5));
+        network = Network.newNetwork();
+        helmChartContainer = new HelmChartContainer().withNetwork(network);
+        helmChartContainer.start();
+        client = helmChartContainer.getKubernetesClient();
+        logWaiter = helmChartContainer.getLogWaiter();
+    }
+
+    @AfterAll
+    @Timeout(value = 5, unit = TimeUnit.MINUTES)
+    static void baseAfterAll() {
+        helmChartContainer.stop();
+        network.close();
+        Awaitility.reset();
+    }
 
     @BeforeEach
     @Timeout(value = 5, unit = TimeUnit.MINUTES)
-    final void baseSetup() {
-        client = helmChartContainer.getKubernetesClient();
-        helmChartContainer.createNamespace(namespace);
+    final void baseSetUp() throws Exception {
+        if (createOperatorNamespace()) {
+            helmChartContainer.createNamespace(operatorNamespace);
+        }
+        if (createPlatformNamespace()) {
+            helmChartContainer.createNamespace(platformNamespace);
+        }
+        if (installOperatorChart()) {
+            helmChartContainer.installOperatorChart(OPERATOR_RELEASE_NAME, "--namespace", operatorNamespace);
+        }
     }
 
     @AfterEach
     @Timeout(value = 5, unit = TimeUnit.MINUTES)
     final void baseTearDown() throws Exception {
-        if (cleanupPlatformChart()) {
-            helmChartContainer.uninstallRelease(PLATFORM_RELEASE_NAME, namespace);
+        if (uninstallPlatformChart()) {
+            helmChartContainer.uninstallRelease(PLATFORM_RELEASE_NAME, platformNamespace, true);
         }
-        if (cleanupNamespace()) {
-            helmChartContainer.deleteNamespace(namespace);
-        } else {
-            // assert that the test cleaned up the namespace on its own
-            assertThat(client.namespaces().withName(namespace).get()).isNull();
+        if (uninstallOperatorChart()) {
+            helmChartContainer.uninstallRelease(OPERATOR_RELEASE_NAME, operatorNamespace, true);
         }
-        if (cleanupOperatorChart()) {
-            helmChartContainer.uninstallRelease(OPERATOR_RELEASE_NAME, "default");
-        }
-        network.close();
     }
 
-    protected boolean cleanupOperatorChart() {
+    protected void installOperatorChartAndWaitToBeRunning(final @NotNull String valuesResourceFile) throws Exception {
+        installOperatorChartAndWaitToBeRunning("-f", valuesResourceFile);
+    }
+
+    protected void installOperatorChartAndWaitToBeRunning(final @NotNull String... commands) throws Exception {
```

Comment:
These are nice helper methods, we should probably add them to the testsuite as well!

Reply:
Agree - will update this [PR](https://github.com/hivemq/hivemq-k8s-testsuite/pull/54) with the same accordingly.

Reply:
done